tests/Pipeline/PipelineTest.php

```diff
@@ -27,6 +27,21 @@ public function testPipelineBasicUsage()
         unset($_SERVER['__test.pipe.two']);
     }
 
+    public function testPipelineUsageWithObjects()
+    {
+        $result = (new Pipeline(new Illuminate\Container\Container))
+            ->send('foo')
+            ->through([new PipelineTestPipeOne])
+            ->then(function ($piped) {
+                return $piped;
+            });
+
+        $this->assertEquals('foo', $result);
+        $this->assertEquals('foo', $_SERVER['__test.pipe.one']);
+
+        unset($_SERVER['__test.pipe.one']);
```

Comment:
you need to call this in a try, finally, otherwise if the assertion fails, it'll screw up the other tests